src/test/scala/org/constellation/domain/transaction/TransactionChainServiceTest.scala

```diff
@@ -37,17 +39,145 @@ class TransactionChainServiceTest
       last.unsafeRunSync shouldBe LastTransactionRef("", 0L)
     }
 
-    "should return last transaction ref if address is known" in {}
+    "should return last transaction ref if address is known" in {
+      val address = "DAG123"
+      val lastTx = LastTransactionRef("tx123", 1)
+      service.lastTransactionRef.set(Map(address -> lastTx)).unsafeRunSync
+
+      val last = service.getLastTransactionRef(address).unsafeRunSync
+      last shouldEqual lastTx
+    }
+  }
+
+  "getLastAcceptedTransactionRef" - {
+    "should return empty string and 0 ordinal if address is unknown" in {
+      val lastAccepted = service.getLastAcceptedTransactionRef("unknown").unsafeRunSync
+
+      lastAccepted shouldEqual LastTransactionRef("", 0L)
+    }
+
+    "should return last accepted transaction ref if address is known" in {
+      val address = "DAG123"
+      val lastTx = LastTransactionRef("tx123", 1)
+      service.lastAcceptedTransactionRef.set(Map(address -> lastTx)).unsafeRunSync
+      val lastAccepted = service.getLastAcceptedTransactionRef(address).unsafeRunSync
+
+      lastAccepted shouldEqual lastTx
+    }
+  }
+
+  "getLastAcceptedTransactionMap" - {
+    "should return current map of accepted transactions" in {
+      service.getLastAcceptedTransactionMap().unsafeRunSync shouldEqual Map.empty
+
+      val lastAcceptedTxRef = Map("DAG123" -> LastTransactionRef("abc", 1))
+      service.lastAcceptedTransactionRef.set(lastAcceptedTxRef).unsafeRunSync
+      val result = service.getLastAcceptedTransactionMap().unsafeRunSync
+
+      result shouldEqual lastAcceptedTxRef
+    }
+  }
+
+  "acceptTransaction" - {
+    "for dummy transaction" - {
+      "should not modify lastAcceptedTransactionRef" in {
+        val dummyTxEdge = mock[Edge[TransactionEdgeData]]
+        val dummyTx = Transaction(dummyTxEdge, LastTransactionRef.empty, isDummy = true)
+        service.acceptTransaction(dummyTx).unsafeRunSync
+
+        val result = service.lastAcceptedTransactionRef.get.unsafeRunSync
+
+        result shouldEqual Map.empty
+      }
+    }
+    "for normal transaction" - {
+      "should add transaction to lastAcceptedTransactionRef" in {
+        val src = "DAG123"
+        val dst = "DAG456"
+        val txEdge = createTransactionEdge(src, dst, LastTransactionRef.empty, 1L, Fixtures.tempKey)
+        val tx = Transaction(txEdge, LastTransactionRef.empty)
+        service.acceptTransaction(tx).unsafeRunSync
+
+        val result = service.lastAcceptedTransactionRef.get.unsafeRunSync
+
+        result shouldEqual Map(src -> LastTransactionRef(tx.hash, 1L))
+      }
+    }
   }
 
   "createAndSetLastTransaction" - {
     "should return new transaction" in {
```

Comment:
This test checks two different things right now. Returning new transaction and reference correctness. It should be in two separate tests

Reply:
Yes, I actually wasn't sure if it deserves it's own test ;).

Reply:
I've learned to ideally have 1 assertion only per test case